### PR TITLE
[codex] docs: explain k8s proxy benefits

### PR DESCRIPTION
**Pull request by officialasishkumar**
## What has changed?

This PR expands the Kubernetes Proxy API docs to explain why teams should use the in-cluster proxy instead of wiring `keploy enterprise` directly into each application Pod.

The update documents the main user-facing benefits: auto-replay during recording, release-aware replay context, capture-time deduplication, REST and MCP automation surfaces, schema generation and schema-conflict handling, the upcoming smart test set model, and local CI replay. It also clarifies edge cases from the implementation, including low-risk noise handling, re-record suite-linking gates, schema insert-vs-update behavior, and the current status of historical/smart-test-set consolidation.

This PR also adds a Kubernetes architecture overview image to the Kubernetes setup guide.

This PR does not resolve a linked issue.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Documentation update (if none of the other choices apply).

## How Has This Been Tested?

- Ran `git diff --check -- versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md`.
- Ran `git diff --check -- versioned_docs/version-4.0.0/keploy-cloud/kubernetes.md static/img/k8s-arch.png`.
- Ran `vale versioned_docs/version-4.0.0/keploy-cloud/kubernetes.md`; it only reports a pre-existing frontmatter issue outside this PR's changed lines.
- Ran `npm run build` successfully after adding the image; Docusaurus generated static files in `build`.
- Ran `npm run serve -- --host 127.0.0.1 --port 3010` successfully earlier on this branch; Docusaurus served the build at `http://127.0.0.1:3010/docs/`, then the local server was stopped.

The visual asset added by this PR is `static/img/k8s-arch.png`.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md [hunk 1]
```diff
@@ -48,17 +48,89 @@ The same `/record/start`, `/record/stop`, `/test/start`, `/deployments`, and rep
...
+
+This is also why replay belongs in the Kubernetes Proxy instead of a one-off CLI run: the proxy has the cluster, Deployment, release, and recording-session context needed to make replay release-aware. For a new-release flow, the proxy layer is designed to coordinate replaying the traffic captured for the new release together with the historical traffic the service has recorded before, so a release is checked against both the latest behavior and the behavior users already depended on. When the smart test set gets shipped, that historical input becomes the curated smart set: newly recorded traffic plus the latest duplicate-free version of the service's long-lived test coverage.
+
+Each test case is exercised once against a freshly rolled Pod and classified into one of three buckets:
+
+- **Pass:** kept as a real, stable test case.
+- **High-risk failure:** marked as failed (a real regression to investigate).
```
Comment (khareyash05) [inline versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md:64]: do we do this? isnt it just pass or fail

**Review by amaan-bhati — COMMENTED** [review]
Thank you for this deep-dive on the Kubernetes Proxy benefits! The updated section is significantly more useful than the bullet-list version - walking through each capability with concrete detail (auto-replay cadence, noise classification buckets, MCP tool names, CI workflow steps) gives teams exactly the context they need to decide whether to adopt the proxy.

**Issues to be Considered:**

- The heading change removes backtick formatting from `` `keploy enterprise` `` (now rendered as plain text `keploy enterprise directly`). Since this is a CLI command name, keeping it in code formatting helps readers scan the page. Worth restoring.

- Section 5 (Smart test set) is labelled "Upcoming" in the heading but the body text switches between present and future tense in a way that may confuse readers about what is actually available today vs. on the roadmap. Consider adding a short callout box at the top of that section like `> **Status: Not yet active in the current record/start path.`" so it's unambiguous.

- This PR modifies `k8s-proxy-api.md`, which is also the target of PR #845 (the revert PR currently open). These two will produce a merge conflict. Please coordinate with the maintainers on which lands first, or rebase onto the result of #845 before this is merged.

- The new `k8s-arch.png` image is added but not tested in both Docusaurus light and dark themes. The image alt text (`How Keploy works on Kubernetes architecture overview`) is descriptive which is great - just make sure the image itself renders legibly in both color modes.

- The Local CI replay section references `keploy test sandbox` as the command but the earlier docs use `keploy test`. If `sandbox` is a separate subcommand, please add a note pointing readers to the CLI reference so they know where to find the flags.

Overall this is a substantive and well-written addition. Sorting out the merge conflict with the revert PR and clarifying the upcoming feature boundary are the main things to address before this is ready.

**Review by khareyash05 — CHANGES_REQUESTED** [review]
please address

File: versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md (hunk 1 above)
Comment (officialasishkumar) [reply versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md:64]: yes, we do have this.. we added this in report as well that gets generated after running keploy test

**Review by khareyash05 — APPROVED** [review]
LGTM

File: versioned_docs/version-4.0.0/keploy-cloud/kubernetes.md [hunk 2]
```diff
@@ -37,6 +37,10 @@ This guide walks you through creating a local **Kind** Kubernetes cluster and co
 > [!NOTE]
 > This documentation covers **local Kind cluster** setup. Documentation for **hosted cluster** setup is coming soon.
 
+## How Keploy works on Kubernetes
```
Comment (charankamarapu) [inline versioned_docs/version-4.0.0/keploy-cloud/kubernetes.md:40]: the arch picture is missing few things.
1. sidecar should be inside pod
2. Only self hosted arch is present 
3. Secrets missing
4. MITM proxy hope reader doesn't get confused with other product - https://github.com/mitmproxy/mitmproxy
5. why does Kubernetes api-server has test case, mocks mentioned.

File: versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md [hunk 3]
```diff
@@ -48,17 +48,89 @@ The same `/record/start`, `/record/stop`, `/test/start`, `/deployments`, and rep
...
+
+### 3. REST API _and_ MCP server
+
+Keploy exposes automation surfaces at two layers:
+
+- The **Kubernetes Proxy REST API** ([Endpoint reference](#endpoint-reference)) handles in-cluster operations such as starting/stopping recording, kicking off replay, fetching session status, and reading logs or reports. All routes outside `/healthz` and the admission webhook sit behind shared-token Bearer auth.
+- The **Keploy API server MCP endpoint** exposes higher-level tools for AI coding tools, including Claude Code, Cursor, Windsurf, and VS Code. This is how an AI agent in your editor authors test suites, runs replays, and scaffolds CI pipelines without you copy-pasting curl commands.
+
+The MCP surface includes around a dozen tools. The headline ones:
```
Comment (charankamarapu) [inline versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md]: when it comes to just k8sproxy these and integration tests these are not useful. Mentioning them will confuse user.

**Review by charankamarapu — CHANGES_REQUESTED** [review]
Please address the comment rest looks good

File: versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md (hunk 3 above)
Comment (officialasishkumar) [reply versioned_docs/version-4.0.0/running-keploy/k8s-proxy-api.md]: removed it

File: versioned_docs/version-4.0.0/keploy-cloud/kubernetes.md (hunk 2 above)
Comment (officialasishkumar) [reply versioned_docs/version-4.0.0/keploy-cloud/kubernetes.md:40]: updated the image

**Review by AkashKumar7902 — APPROVED** [review]

**Review by charankamarapu — APPROVED** [review]
LGTM
